hibernate-core/src/main/java/org/hibernate/sql/results/graph/collection/internal/SelectEagerCollectionInitializer.java

```diff
@@ -28,6 +28,9 @@ public SelectEagerCollectionInitializer(
 
 	@Override
 	public void resolveInstance(RowProcessingState rowProcessingState) {
+		if ( parentAccess != null && !rowProcessingState.isQueryExecution() && parentAccess.findFirstEntityInitializer().isInitialized() ) {
+			return;
```

Comment:
Shouldn't we at least call `persistenceContext.addNonLazyCollection( collectionInstance );` here?

Reply:
I think it should already be in the PC

Reply:
Can you add a test that uses `maxFetchDepth = 0`, which should trigger this particular scenario? First load the owner entity, then do a query with an entity graph for that collection. The query should make sure the collection becomes initialized.